public/js/views/purchase.jsx

```diff
@@ -55,11 +61,23 @@ export default class Purchase extends Component {
                 {...bindActionCreators(purchaseActions, result.dispatch)}
               />
             );
+          } else if (!result.user.braintreeToken) {
+            console.log('Retreiving Braintree Token');
+            return (
+              <BraintreeToken
+                {...bindActionCreators(userActions, result.dispatch) }
+              />
```

Comment:
I hummed and hawed about whether this needed to be a separate component but I decided I liked it :) What I really like is that the component with the connector (Purchase) becomes the only render() function with if statements in it. We should strive to only put if statements in connectors. This makes it so tests of our connectors will cover all integration of smaller units.